t/unit/tasks/test_canvas.py

```diff
@@ -860,6 +877,73 @@ def chord_add():
             _state.task_join_will_block = fixture_task_join_will_block
             result.task_join_will_block = fixture_task_join_will_block
 
+    def test_from_dict(self):
+        header = self.add.s(1, 2)
+        original_chord = chord(header=header)
+        rebuilt_chord = chord.from_dict(dict(original_chord))
+        assert isinstance(rebuilt_chord, chord)
+
+    def test_from_dict_with_body(self):
+        header = body = self.add.s(1, 2)
+        original_chord = chord(header=header, body=body)
+        rebuilt_chord = chord.from_dict(dict(original_chord))
+        assert isinstance(rebuilt_chord, chord)
+
+    def test_from_dict_deep_deserialize(self):
+        header = body = self.add.s(1, 2)
+        original_chord = chord(header=header, body=body)
+        serialized_chord = json.loads(json.dumps(original_chord))
+        deserialized_chord = chord.from_dict(serialized_chord)
+        assert isinstance(deserialized_chord, chord)
+        for task in deserialized_chord.tasks:
+            assert isinstance(task, Signature)
```

Comment:
```suggestion
        assert all(isinstance(task, Signature) for task in deserialized_chord.tasks)
```

That applies to all other loops of this type.

Reply:
Done, this will land in my next push. I've used the trope throughout this test suite and updated two tests unrelated to this diff to match.